### PR TITLE
[DROOLS-1499] When concurrently get prototype ksession bean, it could be repeated

### PR DESCRIPTION
**Pull request by zhugw**

### CODE REVIEW

**Comment by mswiderski** [comment]
ok to test

**Comment by mswiderski** [comment]
@mariofusco it looks ok to me, wdyt?

**Comment by mswiderski** [comment]
@zhugw could you please file jira issue for this?

**Comment by zhugw** [comment]
https://issues.jboss.org/browse/DROOLS-1499

is this ok?

File: kie-spring/src/test/java/org/kie/spring/tests/KieSpringScopeTest.java [hunk 1]
```diff
@@ -88,6 +89,30 @@ public void testStatefulPrototypeKieSession() throws Exception {
...
+        final ConcurrentHashMap ksessionMap = new ConcurrentHashMap();
+
+        CountDownLatch latch = new CountDownLatch(nThreads);
+
+        for (int i = 0; i < nThreads; i++) {
+            executorService.submit(new Runnable() {
+                @Override
+                public void run() {
+                    KieSession ksession = (KieSession) context.getBean("statefulPrototypeSession");
+                    Object put = ksessionMap.put(ksession.hashCode(), new Object());
```
Comment (mariofusco) [inline kie-spring/src/test/java/org/kie/spring/tests/KieSpringScopeTest.java]: The fix is ok but using the hashCode() as unique key of a ksession isn't. I know this is uncommon but it's not impossible that 2 distinct ksessions have the same hashcode. Please replace ksession.hashCode() with ksession.getIdentifier()

**Pull request merged**